From 9550909112af8b001b63561684c259d1f367a94c Mon Sep 17 00:00:00 2001
From: "duncan.macleod" <duncan.macleod@ligo.org>
Date: Tue, 13 Jun 2023 12:11:13 +0100
Subject: [PATCH 2/2] gstlal.pipeio: fix import of collections.abc.Iterable

closes #132
---
 gstlal/python/pipeio.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gstlal/python/pipeio.py b/gstlal/python/pipeio.py
index 173ceabfe..e375f85ac 100644
--- a/gstlal/python/pipeio.py
+++ b/gstlal/python/pipeio.py
@@ -31,7 +31,7 @@
 # =============================================================================
 #
 
-from collections import Iterable
+from collections.abc import Iterable
 
 
 import numpy
-- 
2.40.0

