From 4e77b6a1b236cb8f93282e749d9a4d212b8c0132 Mon Sep 17 00:00:00 2001
From: Duncan Macleod <macleoddm@cardiff.ac.uk>
Date: Tue, 23 Jul 2024 12:44:48 +0100
Subject: [PATCH] bottle: update new_module for python 3.12

---
 gstlal/python/bottle.py | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/gstlal/python/bottle.py b/gstlal/python/bottle.py
index ddcf524cd..270d2c6c9 100644
--- a/gstlal/python/bottle.py
+++ b/gstlal/python/bottle.py
@@ -35,7 +35,7 @@ if __name__ == '__main__':
     if _cmd_options.server and _cmd_options.server.startswith('gevent'):
         import gevent.monkey; gevent.monkey.patch_all()
 
-import base64, cgi, email.utils, functools, hmac, imp, itertools, mimetypes,\
+import base64, cgi, email.utils, functools, hmac, itertools, mimetypes,\
         os, re, subprocess, sys, tempfile, threading, time, warnings
 
 from datetime import date as datedate, datetime, timedelta
@@ -92,6 +92,7 @@ if py3k:
     urlunquote = functools.partial(urlunquote, encoding='latin1')
     from http.cookies import SimpleCookie
     from collections.abc import MutableMapping as DictMixin
+    from types import ModuleType as new_module
     import pickle
     from io import BytesIO
     from configparser import ConfigParser
@@ -109,6 +110,7 @@ else: # 2.x
     from Cookie import SimpleCookie
     from itertools import imap
     import cPickle as pickle
+    from imp import new_module
     from StringIO import StringIO as BytesIO
     from ConfigParser import SafeConfigParser as ConfigParser
     if py25:
@@ -1778,7 +1780,7 @@ class _ImportRedirect(object):
         ''' Create a virtual package that redirects imports (see PEP 302). '''
         self.name = name
         self.impmask = impmask
-        self.module = sys.modules.setdefault(name, imp.new_module(name))
+        self.module = sys.modules.setdefault(name, new_module(name))
         self.module.__dict__.update({'__file__': __file__, '__path__': [],
                                     '__all__': [], '__loader__': self})
         sys.meta_path.append(self)
-- 
2.39.2

